tui/network_dialog_check.go

```diff
@@ -0,0 +1,216 @@
+// Copyright © 2018 Intel Corporation
+//
+// SPDX-License-Identifier: GPL-3.0-only
+
+package tui
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/VladimirMarkelov/clui"
+	term "github.com/nsf/termbox-go"
+
+	"github.com/clearlinux/clr-installer/controller"
+	"github.com/clearlinux/clr-installer/log"
+	"github.com/clearlinux/clr-installer/model"
+	"github.com/clearlinux/clr-installer/progress"
+)
+
+// NetworkTestDialog is dialog window use to stop all other
+// user interaction while the network configuration is tested
+// for connectivity to the swupd server.
+// Check for:
+// - Working network interface
+// - Use of Proxy (if set)
+// - use of Swupd Mirror (if set)
+type NetworkTestDialog struct {
+	DialogBox *clui.Window
+	onClose   func()
+
+	modelSI       *model.SystemInstall
+	progressBar   *clui.ProgressBar
+	progressMax   int
+	resultLabel   *clui.Label
+	dismissButton *SimpleButton
+}
+
+// Success is part of the progress.Client implementation and represents the
+// successful progress completion of a task by setting
+// the progress bar to "full"
+func (dialog *NetworkTestDialog) Success() {
+	dialog.progressBar.SetValue(dialog.progressMax)
+	clui.RefreshScreen()
```

Comment:
You're refreshing the screen twice (here and in the menu callback), is that intentional?

Reply:
Who doesn't like a Fresh a screen? :)  I can remove one of these and retest.

Reply:
@dorileo We are also doing a refresh after SetValue in the tui/installer code and after SetTitle; this seems to be the BKM pattern, no?